Migrate to use device+profile JSON format

kuba2k2 opened the pull request:
This PR allows to choose the target device's profile from the database available in [tuya-cloudcutter.github.io](https://github.com/tuya-cloudcutter/tuya-cloudcutter.github.io). This allows to select a yet-unsupported, provided that another profile already exists for its firmware version.

I've updated the READMEs, adding info about finding the device firmware version and choosing it that way. Also modified the script to download the profile data, in the case of specifying the name on CLI/in environment vars.

I also removed all existing profiles - the updated script doesn't use them anyway. Added the new LSC profile as well. Also added a list of available custom firmware projects - feel free to remove it if it's not appropriate there.

notkmhn commented:
Is this PR still WIP?

kuba2k2 commented:
It might be not WIP, but was tested by one person only (that I know of). This moves all profiles to the tuya-cloudcutter.github.io repository, so any new profiles created here must be created there as well. If you're ready to use the universal profile format instead of the `profile` file-based one, this could be merged.

notkmhn commented:
Gotcha, I suppose it'd be prudent to do some checks and move any new profiles there as necessary before such a switch. Ideally many will be obsolete in the future anyway :)

kuba2k2 commented:
I've updated the READMEs, adding info about finding the device firmware version and choosing it that way. Also modified the script to download the profile data, in the case of specifying the name on CLI/in environment vars.

I also removed all existing profiles - the updated script doesn't use them anyway. Added the new LSC profile as well. Also added a list of available custom firmware projects - feel free to remove it if it's not appropriate there.

m0eter commented:
I see this MR still needs profiles for flashing custom firmware. Is there any actual need for this when we know the firmware version and chipset of the device?

kuba2k2 commented:
Yes, every firmware version needs its own profile (which contains values related to the specific firmware build) and it's not enough to know the version number. We don't have binaries of every possible firmware version, so we only have a finite number of profiles created.

This PR is to simplify profile choice - when your device doesn't have a profile named with its model number, but there's a profile matching the firmware version.

Legacy cloudcutter didn't list firmware versions anywhere, so it was more of a trial-and-error process.

kuba2k2 commented:
The PR has been rebased on 9efb1c4e403e485afb37d0b4bfb28e3fc59bffbb and force-pushed, to fix merge conflicts, involving removing legacy profiles that have been added to main since then.
